File: C++/Arrays and Hashing/TwoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> ans;
        int n = nums.size();
        unordered_map<int,int> u_map;
        for(int i=0 ; i<n ; i++){
            u_map[nums[i]]=i+1;
        }

        for(int i=0 ; i<n ; i++){
            if(u_map[target - nums[i]] > 0 && (u_map[target - nums[i]]-1)!= i){
                ans.push_back(i);
                ans.push_back(u_map[target - nums[i]] - 1);
                break;
            }
        }

        return ans;
    }
};

File: Python/Arrays and Hashing/TwoSum.py
# Ue this when you need both inde and value in pyhitn for loop
# Use the built-in function enumerate():
#
# for idx, x in enumerate(xs):
#     print(idx, x)

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        temp = {};

        for i,num in enumerate(nums):
            temp[num]=i;

        print(temp)
        for i,num in enumerate(nums):
            if (target-num) in temp and temp[target-num]!=i:
                return [temp[target-num],i];
